src/lib/users/dialogs/inst_user_first.rb

```diff
@@ -0,0 +1,661 @@
+
+    def create_dialog
+      Wizard.CreateDialog if Mode.normal # for testing only
+      Wizard.SetTitleIcon("yast-users")
+      # dialog caption
+      Wizard.SetContents(
+        _("Local Users"),
+        contents,
+        main_help,
+        GetInstArgs.enable_back,
+        GetInstArgs.enable_next || Mode.normal
+      )
+      refresh
+      set_focus
+    end
+
+    def close_dialog
+      Wizard.CloseDialog if Mode.normal
+      Progress.set(@progress_orig)
+    end
+
+    # Takes the first word from full name and proposes a login name which is then used
+    # to relace the current login name in UI
+    def propose_login
+      # get the first name
+      full_name = UI.QueryWidget(Id(:full_name), :Value)
+
+      login = full_name.strip.split(" ", 2).first || ""
+      login = UsersSimple.Transliterate(login).delete("^" + UsersSimple.ValidLognameChars).downcase
+
+      UI.ChangeWidget(Id(:username), :Value, login)
+    end
+
+    def refresh
+      @widgets.values.flatten.each do |w|
+        UI.ChangeWidget(Id(w), :Enabled, @widgets[action].include?(w))
+      end
+    end
+
+    def set_focus
+      widget = @widgets[action].first
+      UI.SetFocus(Id(widget)) if widget
+    end
+
+    attr_reader :action
+
+    def action=(value)
+      @action = value
+      refresh
+    end
+
+    def process_new_user_form
+      # username checks
+      @username = UI.QueryWidget(Id(:username), :Value)
+      if @username.empty?
+        Report.Error(
+          # TRANSLATORS: Error popup
+          _("The new username cannot be blank.\n" \
+            "If you don't want to create a user now, select\n" \
+            "'Skip User Creation'.")
+        )
+        return false
+      end
+
+      if !valid_username?(@username)
+        UI.SetFocus(Id(:username))
+        return false
+      end
+
+      # full name checks
+      @full_name = UI.QueryWidget(Id(:full_name), :Value)
+      error = UsersSimple.CheckFullname(@full_name)
+      if !error.empty?
+        Report.Error(error)
+        UI.SetFocus(Id(:full_name))
+        return false
+      end
+
+      # password checks
+      pw1 = UI.QueryWidget(Id(:pw1), :Value)
+      pw2 = UI.QueryWidget(Id(:pw2), :Value)
+      @use_pw_for_root = UI.QueryWidget(Id(:root_pw), :Value)
+
+      if !valid_password?(@username, pw1, pw2)
+        UI.SetFocus(Id(:pw1))
+        return false
+      end
+      @password = pw1
+
+      return true
+    end
+
+    def create_new_user
+      # save the first user data
+      user_map = {
+        "uid"          => @username,
+        "userPassword" => @password,
+        "cn"           => @full_name
+      }
+      UsersSimple.SetUsers([user_map])
+      UsersSimple.SkipRootPasswordDialog(@use_pw_for_root)
+      if root_dialog_follows || @use_pw_for_root
+        UsersSimple.SetRootPassword(@use_pw_for_root ? @password : "")
+      end
+
+      UsersSimple.SetAutologinUser(
+        UI.QueryWidget(Id(:autologin), :Value) == true ? @username : ""
```

Comment:
`== true` is not neccessary. It is ternary logic, and true is only valid that pass it